apps/prairielearn/src/pages/courseSyncs/courseSyncs.js

```diff
@@ -114,20 +94,31 @@ router.post(
 
     if (req.body.__action === 'pull') {
       const jobSequenceId = await syncHelpers.pullAndUpdate(res.locals);
-      res.redirect(res.locals.urlPrefix + '/jobSequence/' + jobSequenceId);
+      res.redirect(`${res.locals.urlPrefix}/jobSequence/${jobSequenceId}`);
     } else if (req.body.__action === 'status') {
       const jobSequenceId = await syncHelpers.gitStatus(res.locals);
-      res.redirect(res.locals.urlPrefix + '/jobSequence/' + jobSequenceId);
+      res.redirect(`${res.locals.urlPrefix}/jobSequence/${jobSequenceId}`);
     } else if (req.body.__action === 'syncImages') {
-      const result = await sqldb.queryAsync(sql.question_images, {
-        course_id: res.locals.course.id,
-      });
-      let images = result.rows;
+      let images;
       if ('single_image' in req.body) {
-        images = _.filter(result.rows, ['image', req.body.single_image]);
```

Comment:
This was changed from filtering the query with all images to find the image to be synced, to creating a new query that checks if the image itself is found in any question. This moves the logic of filtering to the DB, which I believe makes it faster. This was also changed to a separate `__action` string for consistency with other pages.